src/components/views/elements/MemberEventListSummary.js

```diff
@@ -0,0 +1,219 @@
+        let leaveSummary = '';
+        if (leavers) {
+            leaveSummary = (
+                <span>
+                    {leavers} {remainingLeavers > 0 ? 'and ' + remainingLeavers + ' others ':''}left the room
+                </span>
+            );
+        }
+        return (
+            <span>
+                {joinSummary}{joinSummary && leaveSummary?'; ':''}
+                {leaveSummary}
+            </span>
+        );
+    },
+
+    renderAvatars: function(events) {
+        let avatars = events.slice(0, this.props.avatarsMaxLength).map((e) => {
+            let onClickAvatar = () => {
+                dispatcher.dispatch({
+                    action: 'view_user',
+                    member: e.sender,
+                });
+            };
+            return (
+                <MemberAvatar
+                    key={e.getId()}
+                    member={e.sender}
+                    width={14}
+                    height={14}
+                    onClick={onClickAvatar}
+                />
+            );
+        });
+
+        return (
+            <span>
+                {avatars}
+            </span>
+        );
+    },
+
+    render: function() {
+        let summary = null;
+
+        // Reorder events so that joins come before leaves
+        let eventsToRender = this.props.events;
+
+        // Filter out those who joined, then left
+        let filteredEvents = eventsToRender.filter(
+            (e) => {
+                return eventsToRender.filter(
+                    (e2) => {
+                        return e.getSender() === e2.getSender()
+                            && e.event.content.membership !== e2.event.content.membership;
+                    }
+                ).length === 0;
+            }
+        );
+
+        let joinAndLeft = (eventsToRender.length - filteredEvents.length) / 2;
+        if (joinAndLeft <= 0 || joinAndLeft % 1 !== 0) {
+            joinAndLeft = null;
+        }
+
+        let joinEvents = filteredEvents.filter((ev) => {
+            return ev.event.content.membership === 'join';
+        });
+
+        let leaveEvents = filteredEvents.filter((ev) => {
+            return ev.event.content.membership === 'leave';
+        });
+
+        let fewEvents = eventsToRender.length < this.props.threshold;
+        let expanded = this.state.expanded || fewEvents;
+        let expandedEvents = null;
+
+        if (expanded) {
+            expandedEvents = this.props.renderEvents(this.props.previousEvent, eventsToRender);
+        }
+
+        let avatars = this.renderAvatars(joinEvents.concat(leaveEvents));
+
+        let toggleButton = null;
+        let summaryContainer = null;
+        if (!fewEvents) {
+            summary = this.renderSummary(joinEvents, leaveEvents);
+            toggleButton = (
+                <a onClick={this.toggleSummary} href="javascript:;">{expanded?'collapse':'expand'}</a>
+            );
+            let noun = (joinAndLeft === 1 ? 'user' : 'others');
```

Comment:
so... if 2 users join and leave, and nobody joins without leaving or vice versa... we'll just see "2 others joined and left"?